spec.bs

```diff
@@ -491,19 +510,83 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
       1. If the user preference setting disallows the access to topics from |topicsCallerContext|'s [=topics caller context/caller origin=] or |topicsCallerContext|'s [=topics caller context/top level context domain=], then return.
-      1. Let |topics| be the result from running the [=calculate the topics for caller=] algorithm, with |topicsCallerContext| as input.
-      1. Let |headerStructuredFields| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-lists">List</a>.
+      1. Let |topics| be the result of running the [=calculate the topics for caller=] algorithm, with |topicsCallerContext| as input.
+      1. Let |numVersionsInEpochs| be the result of running the [=get the number of distinct versions in epochs=] algorithm, with |topicsCallerContext| as input.
+      1. Let |topicsInnerList| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-inner-lists">Inner List</a>.
       1. Let |lastVersion| be null.
+      1. Let |serializedTopicsInnerListLength| be 0.
       1. For each |topic| in |topics|:
+          1. Let |digitsInTopic| be the number of digits in |topic|["{{BrowsingTopic/topic}}"].
+          1. Increment |serializedTopicsInnerListLength| by |digitsInTopic|.
+          1. If |topicsInnerList| is not empty, then increment |serializedTopicsInnerListLength| by 2.
           1. Let |topicItem| be a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-integers">Integer</a> with value |topic|["{{BrowsingTopic/topic}}"].
           1. Let |topicParameters| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parameters">Parameters</a>.
-          1. If |lastVersion| is null, or if |lastVersion| does not equal |topic|["{{BrowsingTopic/version}}"], then set |topicParameters|["<code>v</code>"] to |topic|["{{BrowsingTopic/version}}"].
+          1. If |lastVersion| is null, or if |lastVersion| does not equal |topic|["{{BrowsingTopic/version}}"]:
+              1. Increment |serializedTopicsInnerListLength| by the length of |topic|["{{BrowsingTopic/version}}"].
+              1. Increment |serializedTopicsInnerListLength| by 3.
+              1. Set |topicParameters|["<code>v</code>"] to a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-tokens">Token</a> having words |topic|["{{BrowsingTopic/version}}"].
           1. Set |lastVersion| to |topic|["{{BrowsingTopic/version}}"].
           1. Associate |topicParameters| with |topicItem|.
-          1. Insert |topicItem| to |headerStructuredFields|.
+          1. Insert |topicItem| to |topicsInnerList|.
+      1. Let |maxNumberOfEpochs| be 3 (i.e. topics are selected from the last 3 epochs).
+      1. Let |topicMaxLength| be 3 (i.e. [=browsing topics types/topic id=] has maximum 3 digits).
+      1. Let |versionMaxLength| be 13 (i.e. chrome.1:1:11).
```

Comment:
This "i.e." isn't clarifying to me. Why is 13 related to `chrome.1:1:11`? What does this mean for user agents with long names, like, say, `internetexplorer`?

Reply:
Clarified -- switched to rely on the current [=browsing topics types/version=].